pkg/meta/sql.go

```diff
@@ -2304,23 +2304,27 @@ func (m *dbMeta) doUpdateDirStat(ctx Context, batch map[Ino]dirStat) error {
 	return nil
 }
 
-func (m *dbMeta) doSyncDirStat(ctx Context, ino Ino) (*dirStat, error) {
-	stat, err := m.calcDirStat(ctx, ino)
-	if err != nil {
-		return nil, err
+func (m *dbMeta) doSyncDirStat(ctx Context, ino Ino) (*dirStat, syscall.Errno) {
+	stat, st := m.calcDirStat(ctx, ino)
+	if st != 0 {
+		return nil, st
 	}
-	err = m.txn(func(s *xorm.Session) error {
-		_, err := s.Insert(&dirStats{ino, stat.length, stat.space, stat.inodes})
+	err := m.txn(func(s *xorm.Session) error {
+		exist, err := s.Exist(&node{Inode: ino})
+		if err != nil {
+			return err
+		}
+		if !exist {
+			return syscall.ENOENT
+		}
+		_, err = s.Insert(&dirStats{ino, stat.length, stat.space, stat.inodes})
 		if err != nil && strings.Contains(err.Error(), "UNIQUE constraint failed") {
```

Comment:
It's incompatible, actually, we even cannot ensure it will work on future version of SQLite. The error returned by doSyncDirStat should be ignored (log only) by most of callers.